### PR TITLE
Graphql logging

### LINKED ISSUE #1697: impl tracing in graphql, rid printlns

### PR DESCRIPTION
**Pull request by miratepuffin**
### What changes were proposed in this pull request?

1. Fixed tracing within the graphql server
2. Pushed logging throughout raphtory - removing all printlns
3. Fixed a constant properties function in graphql which was pub instead of async

### Why are the changes needed?

- Logging and tracing make it a lot easier to see what is going on in the server

### Does this PR introduce any user-facing change? If yes is this documented?

No

### How was this patch tested?

Current test suite

### Issues

Fixes #1697

### CODE REVIEW

**Review by ljeub-pometry — APPROVED** [review]
looks good, some minor formatting cleanup would be good.

File: python/tests/graphql/misc/test_tracing.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import tempfile
+
+from raphtory import Graph
+from raphtory.graphql import GraphServer, RaphtoryClient
+
+# TODO this doesn't work currently due to how we terminate with Tracing
```
Comment (fabianmurariu) [inline python/tests/graphql/misc/test_tracing.py:6]: Have we fixed this?

Comment (miratepuffin) [reply python/tests/graphql/misc/test_tracing.py:6]: Yes, we can turn this test back on.

File: raphtory-graphql/src/config/otlp_config.rs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+                Ok(tracer_provider) => {
+                    info!(
+                        "Sending traces to {}:{}",
+                        self.otlp_agent_host.clone(),
+                        self.otlp_agent_port.clone()
+                    );
+                    Some(tracer_provider)
+                }
+                Err(e) => {
+                    error!("{}", e.to_string());
```
Comment (fabianmurariu) [inline raphtory-graphql/src/config/otlp_config.rs:63]: do we just move on if we were told to enable tracing but we couldn't?

**Comment by fabianmurariu** [comment]
This was a lot of work, but I can't help but thinking 90% of these `println!` should have been removed instead of replaced

**Pull request merged**